### PR TITLE
Remove Unused Legacy Container Hot Swap

### LINKED ISSUE #3738: NullRuntime should be removed from repo
And all workflows that rely on it should be fixed (likely by forking new work items).
In particular, Replay tool does not work properly - it should not support running from ops, and should always start with snapshot (--from argument).

Would appreciate help here with cleanup!

### LINKED ISSUE #5471: When satisfies() returns false, our telemetry has almost no signal to tell what happened
Per https://portal.microsofticm.com/imp/v3/incidents/details/231266457/home, it's very hard to say what's going on with file when it gets stuck due to code proposal / versioning when satisfies() returns false.
At minimum, we should put more  descripting error to container.close() when we get into this state.

### PR DESCRIPTION
**Pull request by anthony-murphy**
Removes all code related to hotswap, which has never been used, and doesn't work. To support legacy containers created before create detached, i added a wait in the protocol handler to wait until the code details has been set, this allows remove of the null runtime. The snapshot test cover this case. Lastly renamed the simplified reloadContextCore to processCodeProposal and added an error if the container is closed due to the current context not satisfying the incoming context

fixes #5471 
fixes #3738
related #4685

### CODE REVIEW

**Comment by curtisman** [comment]
This should close #3738 too, right?

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 1]
```diff
@@ -892,17 +873,16 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
...
 
         // Create the SummaryManager and mark the initial state
         this.summaryManager = new SummaryManager(
             context,
             this.runtimeOptions.summaryOptions.generateSummaries !== false,
             this.logger,
             (summarizer) => { this.nextSummarizerP = summarizer; },
-            this.previousState.nextSummarizerP,
-            !!this.previousState.reload,
+            undefined,
```
Comment (curtisman) [inline packages/runtime/container-runtime/src/containerRuntime.ts:884]: Remove these two unused parameter from SummaryManager too?

**Comment by ChumpChief** [comment]
Loving this change - also @heliocliu since this will affect his work

File: packages/loader/container-loader/src/container.ts [hunk 2]
```diff
@@ -979,90 +969,21 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
...
-            sequenceNumber: this._deltaManager.lastSequenceNumber,
-            term: this._deltaManager.referenceTerm,
-        };
-
-        await this.loadContext(codeDetails, attributes, snapshot, state.prevState);
-
-        this.deltaManager.inbound.resume();
-        this.deltaManager.inboundSignal.resume();
+        this.close(CreateContainerError(
+            new Error("Existing context does not satisfy incoming code proposal")));
```
Comment (anthony-murphy) [inline packages/loader/container-loader/src/container.ts:986]:
```suggestion
            new Error("ExistingContextDoesNotSatisfyIncomingProposal")));
```

**Review by curtisman — APPROVED** [review]

**Review by ChumpChief — APPROVED** [review]

File: packages/loader/container-loader/src/container.ts [hunk 3]
```diff
@@ -1427,6 +1342,26 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             this.connectionStateHandler.receivedRemoveMemberEvent(clientId);
         });
 
+        // back compat for old containers where the code details
```
Comment (vladsud) [inline packages/loader/container-loader/src/container.ts:1345]: @anthony-murphy , do we actually need this?
I believe this is only needed if there is no snapshot created for old file (or someone decided to load from ops only), right?
Reason I ask - I'd really love to remove code in Contaier.load() that connects to web socket if there is no snapshot. It's is confusing too many people :) If file was never summarized, I'd consider it broken / abandoned at that moment

**Pull request merged**